trace_decoder/src/lib.rs

```diff
@@ -1,14 +1,141 @@
+//! provides a use case for this library:
+//! ```ignore
+//!  pub async fn prove(
+//!      // In this example, [self] is a [ProverInput] storing a [BlockTrace] and
+//!      // [OtherBlockData].
+//!      self,
+//!      runtime: &Runtime,
+//!      previous: Option<PlonkyProofIntern>,
+//!  ) -> Result<GeneratedBlockProof> {
+//!      let block_number = self.get_block_number();
+//!      info!("Proving block {block_number}");
+//!
+//!      let other_data = self.other_data;
+//!      // The method calls [into_txn_proof_gen_ir] (see below) to
+//!      // generate an IR for eaach block transaction.
+//!      let txs = self.block_trace.into_txn_proof_gen_ir(
+//!          &ProcessingMeta::new(resolve_code_hash_fn),
+//!          other_data.clone(),
+//!      )?;
+//!
+//!      // The block IRs are provided to the prover to generate an
+//!      // aggregation proof.
+//!      let agg_proof = IndexedStream::from(txs)
+//!          .map(&TxProof)
+//!          .fold(&AggProof)
+//!          .run(runtime)
+//!          .await?;
+//!
+//!      
+//!      if let AggregatableProof::Agg(proof) = agg_proof {
+//!          let prev = previous.map(|p| GeneratedBlockProof {
+//!              b_height: block_number.as_u64() - 1,
+//!              intern: p,
+//!          });
+//!
+//!          // The final aggregation proof is then used to prove the
+//!          // current block.
+//!          let block_proof = Literal(proof)
+//!              .map(&BlockProof { prev })
+//!              .run(runtime)
+//!              .await?;
+//!
+//!          info!("Successfully proved block {block_number}");
+//!          Ok(block_proof.0)
+//!      } else {
+//!          bail!("AggProof is is not GeneratedAggProof")
+//!      }
+//!  }
+//! ```
+//!
+//! As we see in the example, to turn a [BlockTrace] into a
+//! vector of IRs, one must call the method
+//! [into_txn_proof_gen_ir](BlockTrace::into_txn_proof_gen_ir):
+//! ```ignore
+//! pub fn into_txn_proof_gen_ir<F>(
+//!     self,
+//!     // Specifies the way code hashes should be dealt with.
+//!     p_meta: &ProcessingMeta<F>,
+//!     // Extra data needed for proof generation.
+//!     other_data: OtherBlockData,
+//! ) -> TraceParsingResult<Vec<TxnProofGenIR>>
+//! ```
+//!
+//! It first preprocesses the [BlockTrace] to provide transaction,
+//! withdrawals and tries data that can be directly used to generate an IR.
+//! [into_processed_block_trace](BlockTrace::into_processed_block_trace) is the
+//! method that produces such a [ProcessedBlockTrace].
+//! For each transaction,
+//! [into_txn_proof_gen_ir](BlockTrace::into_txn_proof_gen_ir) extracts the
+//! necessary data from the [ProcessedTxnInfo] to
+//! return the IR.
+//!
+//! The IR is used to generate root proofs, then aggregation proofs and finally
+//! block proofs. But an aggregation proof requires at least two entries. We
+//! ensure this by padding the vector of IRs thanks to additional dummy
```

Comment:
nit:
```suggestion
//! block proofs. Because aggregation proofs require at least two entries, we
//! pad the vector of IRs thanks to additional dummy payload
```